presto-main/src/main/java/com/facebook/presto/operator/LookupJoinOperator.java

```diff
@@ -238,6 +240,18 @@ private boolean tryFetchLookupSourceProvider()
         return true;
     }
 
+    private boolean isLookupSourceEmpty()
+    {
+        return !probeOnOuterSide &&
+                ((closed && joinPosition == -1) ||
+                        (!finishing &&
```

Comment:
This is unreadable. If you put the logic in `tryFetchLookupSourceProvider`, there are preconditions to check (no `closed`, no `joinPosition`, no `finishing`)